core/app/models/spree/payment/processing.rb

```diff
@@ -40,7 +40,12 @@ def capture!(amount=nil)
 
           money = ::Money.new(amount, currency)
           capture_events.create!(amount: money.to_f)
-          handle_response(response, :complete, :failure)
+
+          if uncaptured_amount > 0
+            handle_response(response, :pend, :failure)
+          else
+            handle_response(response, :complete, :failure)
+          end
```

Comment:
You have a bit of duplication there. If you extract this logic to a function

```
uncaptured_amount > 0 ? :pend : :complete
```

Then you can just call

```
handle_response(response, whatever_you_called_that_function, :failure)
```